mne_bids/dataframe_func.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+
+
+def odict_to_ndarray(d):
+    """ Cast an ordered dictionary to a numpy ndarray """
+    titles = np.reshape(np.array(list(d.keys())), (1, len(d)))
+    data = np.stack(d.values(), axis=-1).astype(str)
+    return np.concatenate([titles, data])
+
+
+def from_tsv(fname):
+    """ Read a tsv file into an ndarray """
+    return np.loadtxt(fname, dtype=str, delimiter='\t')
+
+
+def to_tsv(data, fname):
+    """ Write an ndarray to a tsv file """
+    np.savetxt(fname, data, fmt='%s', delimiter='\t', comments='')
+
+
+def is_contained(data, other):
+    """ Checks whether `other` is contained within `data`
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+    other : numpy.ndarray
+
+    """
+    for row in other[1:]:
+        if not row.tolist() in data[1:].tolist():
+            return False
+    return True
+
+
+def column_data(data, column):
+    """ Returns the data in the specified column """
+    col_num = data[0].tolist().index(column)
+    return data[1:, col_num]
+
+
+def combine_data(data, other, drop_column=None):
+    """ Add two ndarrays of data with column names together
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Original array of data.
+    other : numpy.ndarray
+        New array of data.
+    drop_column : str
+        Name of the column to check for duplicate values in.
+        Any duplicates found will be dropped from the original data array (ie.
+        most recent value are kept).
+
+    """
+    data_head, data_data = np.split(data, [1])
+    other_head, other_data = np.split(other, [1])
```

Comment:
have you given some thought how to store the column headers? namedtuple? dictionary? let's try not to recreate the complexity that comes with pandas dataframes ...

Reply:
basically you don't want to mix the label (column header in pandas) with the rest because the label is string whereas the rest could be any other data type